app/src/main/java/io/intrepid/webservicesexercise/view/MainActivity.java

```diff
@@ -0,0 +1,61 @@
+package io.intrepid.webservicesexercise.view;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.Toast;
+import com.squareup.picasso.Picasso;
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import io.intrepid.webservicesexercise.R;
+import io.intrepid.webservicesexercise.presenter.AppPresenter;
+import io.intrepid.webservicesexercise.presenter.Presenter;
+import timber.log.Timber;
+
+public class MainActivity extends AppCompatActivity implements View {
+
+    private static final String TAG = MainActivity.class.getSimpleName();
+    @BindView(R.id.username_input)
+    EditText usernameInputView;
+    @BindView(R.id.avatar)
+    ImageView avatarView;
+    private Presenter presenter;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        presenter = new AppPresenter(this);
+        ButterKnife.bind(this);
+        Timber.plant(new Timber.DebugTree());
+    }
+
+    @OnClick(R.id.submit_button)
+    public void submitButtonClicked() {
+        if (usernameInputView.getText().length() > 0) {
```

Comment:
to keep the view's responsibility as limited as possible, you should pass the input string to the presenter, whether it's empty or not; the presenter will then perform the logic as to whether the string is empty -- if it isn't, it would perform the web request, else it would call a method on View, e.g., showEmptyInputMessage(@StringRes int messageId) and in the View's implementation of that method, it would execute `Toast.makeText(this, getString(R.string.empty_search_prompt), Toast.LENGTH_SHORT).show();`